tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/IFindReplaceUIAccess.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.ui.workbench.texteditor.tests;
+
+import java.util.Set;
+
+import org.eclipse.swt.widgets.Widget;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+
+import org.eclipse.ui.internal.findandreplace.IFindReplaceLogic;
+import org.eclipse.ui.internal.findandreplace.SearchOptions;
+
+
+/**
+ * Wraps UI-Access for different Find/Replace-UIs
+ */
+interface IFindReplaceUIAccess {
+
+	IFindReplaceTarget getTarget();
+
+	void close();
+
+	void ensureHasFocusOnGTK();
+
+	void unselect(SearchOptions option);
+
+	void select(SearchOptions option);
+
+	void simulateEnterInFindInputField(boolean shiftPressed);
+
+	void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed);
+
+	String getFindText();
+
+	String getReplaceText();
+
+	void setFindText(String text, boolean grabFocus);
+
+	void setReplaceText(String text);
+
+	Widget getButtonForSearchOption(SearchOptions option);
+
+	Set<SearchOptions> getEnabledOptions();
+
+	Set<SearchOptions> getSelectedOptions();
+
+	IFindReplaceLogic getFindReplaceLogic();
+
+	void pressReplaceAll();
```

Comment:
There are the methods `pressReplaceAll()` and `performReplace()`. I would prefer to have a uniform naming, either referring to a button press or to performing some action (by doing a button press).

Reply:
Since "single replace" does more than just press a button and overally, what really matters is that the action is performed, I have renamed both to "perform..."

https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa